lhotse/features/base.py

```diff
@@ -593,12 +593,13 @@ def data(self) -> Union[Dict[str, Features], Iterable[Features]]:
         """Alias property for ``self.features``"""
         return self.features
 
-    def to_eager(self) -> "FeatureSet":
-        return FeatureSet(list(self))
-
     @staticmethod
-    def from_features(features: Iterable[Features]) -> "FeatureSet":
-        return FeatureSet(list(features))  # just for consistency with other *Sets
+    def from_features(features: Union[Iterable[Features], LazyMixin]) -> "FeatureSet":
+        return (
+            FeatureSet([f for f in features])
+            if isinstance(features, LazyMixin)
```

Comment:
Curious why not use one of these two options for both cases? I recall list has some weird behavior with regard to whether len is implemented, but maybe we can just always use comprehension?

Reply:
Yeah, `list()` requires `__len__` to be implemented, which is why I had to use list comprehension. However, it seems like list comprehension may be slower than list() (see [this](https://stackoverflow.com/questions/29356846/python-list-vs-list-comprehension-building-speed)), which is why I didn't make it the default.